File: tasks/2026-05-12-long-term-project/album-tracker/mark_covers_unavailable.js
const sql = require('sql.js');
const fs = require('fs');

const DB_PATH = 'C:/Users/qujt/.qclaw/workspace/_music_latest.db';

// Debug: check if file exists
if (!fs.existsSync(DB_PATH)) {
    console.error('Database not found at:', DB_PATH);
    process.exit(1);
}

// Load database
const fileBuffer = fs.readFileSync(DB_PATH);
const db = new sql.Database(fileBuffer);

// Get albums without covers
const result = db.exec(`
    SELECT album_id, album_name, artist 
    FROM albums 
    WHERE cover_image_url IS NULL OR cover_image_url = "" 
    ORDER BY total_listen_count DESC, rating DESC
`);

if (result.length === 0 || !result[0].values) {
    console.log('No albums without covers found');
    process.exit(0);
}

const albums = result[0].values;
console.log(`Total albums without covers: ${albums.length}`);
console.log('='.repeat(80));

// Mark each as UNAVAILABLE
const updateStmt = db.prepare('UPDATE albums SET cover_image_url = "UNAVAILABLE" WHERE album_id = ?');
const yearlyTables = ['albums_2024', 'albums_2025', 'albums_2026'];

for (const [albumId, albumName, artist] of albums) {
    console.log(`${albumId}: ${artist} - ${albumName}`);
    
    // Update main table
    updateStmt.run([albumId]);
    
    // Update yearly tables if they exist
    for (const table of yearlyTables) {
        try {
            db.run(`UPDATE ${table} SET cover_image_url = "UNAVAILABLE" WHERE album_id = ?`, [albumId]);
        } catch (e) {
            // Table might not exist or album not in table
        }
    }
}

updateStmt.free();

// Save database
const data = db.export();
fs.writeFileSync(DB_PATH, Buffer.from(data));
console.log('='.repeat(80));
console.log(`Marked ${albums.length} albums as UNAVAILABLE`);
console.log('Database saved to', DB_PATH);

db.close();

File: tasks/2026-05-12-long-term-project/album-tracker/mark_covers_unavailable.py
import sqlite3
import sys
import io

# Fix Windows encoding issue\sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')

# Connect to database
conn = sqlite3.connect('_music_latest.db')
c = conn.cursor()

# Get albums without covers
c.execute('SELECT album_id, album_name, artist FROM albums WHERE cover_image_url IS NULL OR cover_image_url = "" ORDER BY total_listen_count DESC, rating DESC')
rows = c.fetchall()

print(f'Total albums without covers: {len(rows)}')
print('=' * 80)

# Update each album to mark as unavailable
for row in rows:
    album_id = row[0]
    album_name = row[1]
    artist = row[2]
    
    print(f'{album_id}: {artist} - {album_name}')
    
    # Mark as unavailable by setting a special marker
    c.execute('UPDATE albums SET cover_image_url = "UNAVAILABLE" WHERE album_id = ?', (album_id,))
    
    # Also update in yearly tables if they exist
    for table in ['albums_2024', 'albums_2025', 'albums_2026']:
        try:
            c.execute(f'UPDATE {table} SET cover_image_url = "UNAVAILABLE" WHERE album_id = ?', (album_id,))
        except:
            pass  # Table might not exist or album not in table

conn.commit()
print('=' * 80)
print(f'Marked {len(rows)} albums as UNAVAILABLE')

conn.close()
